nilearn/typing.py

```diff
@@ -27,46 +27,46 @@
 from numpy import ndarray
 from numpy.typing import DTypeLike
 
-BorderSize = int
-Connected = int
+Connected = bool
 Detrend = bool
-LowerCutoff = float
-MemoryLevel = int
-NJobs = int
-NPerm = int
 Resume = bool
 Standardize = bool
 Tfce = bool
 TwoSidedTest = bool
-UpperCutoff = float
-Verbose = int
-
 
 # TODO update when dropping python 3.9
 if sys.version_info[1] < 10:
+    BorderSize = (int, np.integer)
```

Comment:
I think the types parameter templates in nilearn/_utils.doc.py should be adapted to match, no?

@man-shu @bthirion

Reply:
<img width="489" alt="Screenshot 2025-02-10 at 9 51 58 AM" src="https://github.com/user-attachments/assets/6a522ecc-3802-4e19-ac87-a90f31e15621" />

This is what's there in `nilearn/_utils.doc.py` right now. Do you mean we should add np.integer too here?

Reply:
I feel like the difference between the two is too technical (explained [here](https://stackoverflow.com/a/38156630)) and `int` semantically covers all of them.

Reply:
yeah I was not sure about how 'pedantic" we want our doc string to be